### PR TITLE
mock emit and added verify mocks

### PR DESCRIPTION
**Pull request by noambergIL**

### CODE REVIEW

File: go/fake/fake_sdk.go [hunk 1]
```diff
@@ -94,11 +107,20 @@ func (m *mockHandler) SdkAddressGetCallerAddress(ctx context.ContextId, permissi
 	return m.callerAddress
 }
 
-
 func (m *mockHandler) SdkEventsEmitEvent(ctx context.ContextId, permissionScope context.PermissionScope, eventFunctionSignature interface{}, args ...interface{}) {
-	panic("implement me")
-}
+	var key []interface{}
+	//	key = append(key, eventFunctionSignature) // is there a way to equal funcs?
```
Comment (electricmonk) [inline go/fake/fake_sdk.go]: please delete commented out code

File: go/fake/fake_sdk.go [hunk 2]
```diff
@@ -48,8 +58,10 @@ func (m *mockHandler) SdkServiceCallMethod(ctx context.ContextId, permissionScop
 	key = append(key, serviceName, methodName)
 	key = append(key, args...)
 
-	for _, stub := range m.serviceStubs {
+	for i, stub := range m.serviceStubs {
 		if keyEquals(stub.key, key) {
+			m.serviceStubs[i].satisfied = true
```
Comment (electricmonk) [inline go/fake/fake_sdk.go]: why both indexed access and pointer access?

File: go/fake/fake_sdk.go [hunk 3]
```diff
@@ -120,6 +142,31 @@ func (m *mockHandler) MockServiceCallMethod(serviceName string, methodName strin
...
 }
 
+func (m *mockHandler) MockEmitEvent(eventFunctionSignature interface{}, args ...interface{}) {
+	var key []interface{}
+	//	key = append(key, eventFunctionSignature) // is there a way to equal funcs?
+	key = append(key, args...)
+	m.eventStubs = append(m.eventStubs, eventStub{key: key})
+}
+
+func (m *mockHandler) VerifyMocks() {
```
Comment (electricmonk) [inline go/fake/fake_sdk.go]: question: should we call `VerifyMocks` after the test finished running automatically? this could be achieved in the "test framework" (the `inscope` function)

File: go/fake/fake_sdk_test.go [hunk 4]
```diff
@@ -147,3 +146,105 @@ func TestMockHandler_SdkEventsEmitEvent_Unstubbed(t *testing.T) {
...
+	arg2 := 1
+	out := []interface{}{true, "z"}
+
+	s.MockServiceCallMethod(serviceName, methodName, out, arg1, arg2)
+	s.SdkServiceCallMethod(0, 0, serviceName, methodName, arg1, arg2)
+
+	require.NotPanics(t, func() { s.VerifyMocks() })
+}
+
+func TestMockHandler_Verify_CallMethodMissing(t *testing.T) {
```
Comment (electricmonk) [inline go/fake/fake_sdk_test.go]: why did you add new tests rather than add `VerifyMocks` to existing tests? you're essentially duplicating all tests here

**Comment by noambergIL** [comment]
fixed stuff please approve
